src/components/SearchModal/CurrencySearch.tsx

```diff
@@ -347,12 +348,17 @@ export function CurrencySearch({
     fetchFavoriteTokenFromAddress()
   }, [fetchFavoriteTokenFromAddress])
 
-  const fetchingToken = useRef(false)
+  const fetchingToken = useRef<string | null>(null)
+
+  useEffect(() => {
+    fetchingToken.current = null
+  }, [chainId, debouncedQuery, defaultTokens])
 
   const fetchListTokens = useCallback(
     async (page?: number) => {
       if (fetchingToken.current) return
-      fetchingToken.current = true
+      const fetchId = uuid()
```

Comment:
I think we can use `Date.now()` here

Reply:
We already have uuid lib , so using it is still fine